### PR TITLE
Fix Railgun crash and minor projectile optimizations

### LINKED ISSUE #16477: Railguns can crash on certain firing angles.
See fixes in https://github.com/IceReaper/KKnD/pull/26 and
https://github.com/AttacqueSuperior/Engine/commit/3ad4e68b204eee35bbcb5dfc2ecf08f01bdca0b8 which reuse the same vector calculation.

### PR DESCRIPTION
**Pull request by reaperrr**
Fixes #16477 and does some minor perf/style/clean-up optimizations on several other projectiles.

### CODE REVIEW

File: OpenRA.Mods.Common/Projectiles/Railgun.cs [hunk 1]
```diff
@@ -153,14 +153,17 @@ void CalculateVectors()
 
 			// An easy vector to find which is perpendicular vector to forwardStep, with 0 Z component
 			LeftVector = new WVec(ForwardStep.Y, -ForwardStep.X, 0);
-			LeftVector = 1024 * LeftVector / LeftVector.Length;
+			if (LeftVector.Length != 0)
```
Comment (pchote) [inline OpenRA.Mods.Common/Projectiles/Railgun.cs]: Using `LengthSquared` for these will save an unnecessary square root

**Comment by reaperrr** [comment]
From IRC:
> [01:10] <+pchote> reaperrr: can you put the sync attributes on their own line to reduce rebase conflicts with my stylecop pr?

File: OpenRA.Mods.Common/Projectiles/Railgun.cs (hunk 1 above)
Comment (reaperrr) [reply OpenRA.Mods.Common/Projectiles/Railgun.cs]: Done.

**Comment by reaperrr** [comment]
Updated.

> [01:10] <+pchote> reaperrr: can you put the sync attributes on their own line to reduce rebase conflicts with my stylecop pr?

To be honest, as far as I can tell in most places that would either have been counterproductive (in lines that auto-resolve should be able to handle), or wouldn't help (lines that would still conflict for different reasons, like inserted empty lines or removed [Sync] attribute etc.).

If you think there are more places that would really not conflict anymore if I updated them, please point them out.

**Comment by pchote** [comment]
I was thinking more that I could then drop these files completely from #16532 and make that PR depend on this one.

**Comment by reaperrr** [comment]
OK, updated.

**Review by pchote — APPROVED** [review]

**Review by teinarss — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/de38ed83e427b6f1cb769fc15f09f6a48e20d846...473c64bf005f5192e4a3925578865c17d3aa5408

**Pull request merged**